Library/TelemetryTypes/TelemetryType.ts

```diff
@@ -0,0 +1,13 @@
+/**
+ * Telemetry type enumeration
+ */
+enum TelemetryType {
```

Comment:
FYI - TypeScript 2.4+ allows string-based enums. That could avoid some of the map-type syntax you use in the envelope factory.

Reply:
Updated and changed.